lib/broccoli/ember-app.js

```diff
@@ -634,6 +635,39 @@ class EmberApp {
     }, []);
   }
 
+  _compileAddonTemplates(tree) {
+    tree = preprocessTemplates(tree, {
+      annotation: `_compileAddonTemplates`,
+      registry: this.registry,
+    });
+
+    return tree;
+  }
+
+  _compileAddonJs(tree) {
+    let precompiledSource = new AmdExcluder(tree, {
+      annotation: `_compileAddonJs`,
+    });
+
+    precompiledSource = preprocessJs(precompiledSource, '/', '/', {
+      annotation: `_compileAddonJs`,
+      registry: this.registry,
+    });
```

Comment:
I'm wondering if in terms of performance it would be better to teach `preprocessJs()` about the AMD exclusion rule 🤔    but then again that is something that can probably be refactored later...